Core/MMKV_IO.cpp

```diff
@@ -574,12 +576,25 @@ bool MMKV::setDataForKey(MMBuffer &&data, MMKVKey_t key, bool isDataHolder) {
                 isDataHolder = false;
             }
         }
+
+        bool onlyOneKey = !m_isInterProcess && m_dicCrypt->size() == 1 && m_dicCrypt->find(key) != m_dicCrypt->end();
+
         auto itr = m_dicCrypt->find(key);
         if (itr != m_dicCrypt->end()) {
```

Comment:
This is a second `find()` & check after `onlyOneKey`. You should probably only `find()` once.